### PR TITLE
Flesh piercer drop rate

### LINKED ISSUE #5194: Flesh Piercer [3336] Does Not Drop
## CURRENT BEHAVIOUR:

The Flesh Piercer ([Wowgaming](https://wowgaming.altervista.org/aowow/?item=3336), [Wowhead Classic](https://classic.wowhead.com/item=3336/flesh-piercer), [Wowhead TBC](https://tbc.wowhead.com/item=3336/flesh-piercer)) does not currently drop from [Daggerspine Shorestalkers](https://wowgaming.altervista.org/aowow/?npc=2368) or [Daggerspine Screamers](https://wowgaming.altervista.org/aowow/?npc=2370) as intended.

## EXPECTED BLIZZLIKE BEHAVIOUR:

The drop rate, so far as I can tell from the data from both the Classic and TBC Wowhead databases, should be 1.5% for Shorestalkers and 1.1% for Screamers. If these values are not acceptable, they should at least equal the provided values from Wowgaming, and other WotLK databases, which is 0.1%.

## SOURCE:

image

## STEPS TO REPRODUCE THE PROBLEM:

1. Run AzerothCore
2. Head to Hillsbrad Foothills
image
3. Kill Daggerspine Shorestalkers and Screamers
4. Never receive the Flesh Piercer

## EXTRA NOTES:

This item exists in two places in the database, `item_template` (where the item is defined) and `reference_loot_template` (as the screenshot above shows). The item lives in a reference loot table that is shared with many more creatures than just Daggerspine Shorestalkers and Screamers, and thus, should be removed (since it isn't even dropping anyways).

## AC HASH/COMMIT:

54e371e0e7fa2911eb02ff8a3c042d51e7d4ef5b

## OPERATING SYSTEM:

Ubuntu 18

## MODULES:

None

## CUSTOMIZATIONS:

None

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/97710392-flesh-piercer-3336-does-not-drop?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by ReynoldsCahoon**
## Changes Proposed:
-  Remove bogus entry in `reference_loot_template` for Flesh Piercer
-  Add proper entries in `creature_loot_template` for Flesh Piercer

## Issues Addressed:
- Closes #5194

## SOURCE:

[Wowhead Classic](https://classic.wowhead.com/item=3336/flesh-piercer)
[Wowhead TBC](https://tbc.wowhead.com/item=3336/flesh-piercer)

## Tests Performed:

- I added these values to my database
- Set `Rate.Drop.Item.Uncommon = 10`
- Went to Hillsbrad Foothills
- Killed ~10-15 of these and the item dropped.

## How to Test the Changes:

- Follow the above steps, and go to this location:
image
- Kill Daggerspine Shorestalkers and Daggerspine Screamers.

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by pklloveyou** [comment]
Tested

File: data/sql/updates/pending_db_world/rev_1617902440529783360.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1617902440529783360');
+
+# Remove Flesh Piercer from Reference Table
+DELETE FROM `reference_loot_template` WHERE `Entry` = 24068 AND `Item` = 3336;
+
+# Adds Flesh Piercer to Daggerspine Shorestalkers and Screamers loot tables
```
Comment (pklloveyou) [inline data/sql/updates/pending_db_world/rev_1617902440529783360.sql:6]:
```sql
DELETE FROM `creature_loot_template` WHERE `Entry` = 2368 AND `Item` = 3336;
DELETE FROM `creature_loot_template` WHERE `Entry` = 2370 AND `Item` = 3336;
-- Adds Flesh Piercer to Daggerspine Shorestalkers and Screamers loot tables
```

File: data/sql/updates/pending_db_world/rev_1617902440529783360.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1617902440529783360');
+
+# Remove Flesh Piercer from Reference Table
```
Comment (pklloveyou) [inline data/sql/updates/pending_db_world/rev_1617902440529783360.sql:3]: -- Remove Flesh Piercer from Reference Table

**Comment by ReynoldsCahoon** [comment]
Ah, sorry for the nonstandard comments. Thanks for addressing.

**Comment by ReynoldsCahoon** [comment]
Is it best practice to delete the branch now that this is merged?

**Comment by ErnestoGonzalezC** [comment]
https://stackoverflow.com/questions/10765321/should-i-delete-a-branch-after-merging-it#:~:text=4%20Answers&text=There's%20no%20problem%20in%20deleting,after%20the%20PR%20got%20accepted).

**Comment by ErnestoGonzalezC** [comment]
> Is it best practice to delete the branch now that this is merged?

https://stackoverflow.com/questions/10765321/should-i-delete-a-branch-after-merging-it#:~:text=4%20Answers&text=There's%20no%20problem%20in%20deleting,after%20the%20PR%20got%20accepted).

**Pull request merged**